### PR TITLE
Fix cv2

### PR DESCRIPTION
**Pull request by tangy5**
remove cv2 usage base.

use skimage find contour, note skimage uses subpixel interpolation for contour detection, boundaries can be subpixel-wise (0.5 accurate). 

Unit tests and local tests passed. 

One unit test case is changed because of the subpixel accuracy of skimage changed from cv2.

### CODE REVIEW

File: sample-apps/pathology/lib/utils.py [hunk 1]
```diff
@@ -556,9 +568,30 @@ def _group_item(groups, d, output_dir):
...
+def calculate_bounding_rect(points):
+    points = np.array(points, dtype=int)
+    x_min, y_min = np.min(points, axis=0)
+    x_max, y_max = np.max(points, axis=0)
+    w = x_max - x_min + 1
+    h = y_max - y_min + 1
+    return int(x_min), int(y_min), int(w), int(h)
+
+
+def fill_poly(image_size, polygons, color, mode="L"):
```
Comment (KumoLiu) [inline sample-apps/pathology/lib/utils.py:580]: Didn't see where use this function.

File: monailabel/transform/post.py [hunk 2]
```diff
@@ -220,43 +214,78 @@ def __call__(self, data):
...
-                                polygons.append(coords)
-                                break
-                    else:
-                        polygons.append(coords)
-
-                if len(polygons):
-                    logger.debug(f"+++++ {label_idx} => Total Polygons Found: {len(polygons)}")
-                    elements.append({"label": label_name, "contours": polygons})
+                try:
+                    import cv2
```
Comment (SachidanandAlle) [inline monailabel/transform/post.py:227]: use monai util optional import..

File: sample-apps/pathology/lib/trainers/hovernet_nuclei.py [hunk 3]
```diff
@@ -71,13 +75,24 @@ def _fetch_datalist(self, request, datastore: Datastore):
             img = np.array(Image.open(d["image"]).convert("RGB"))
             ann_type = np.array(Image.open(d["label"]))
 
-            numLabels, ann_inst, _, _ = cv2.connectedComponentsWithStats(ann_type, 4, cv2.CV_32S)
+            try:
+                import cv2
```
Comment (SachidanandAlle) [inline sample-apps/pathology/lib/trainers/hovernet_nuclei.py:79]: same here.. refer: https://github.com/Project-MONAI/MONAILabel/blob/main/sample-apps/radiology/lib/configs/segmentation.py#L26

**Review by SachidanandAlle — APPROVED** [review]

File: requirements.txt [hunk 4]
```diff
@@ -28,7 +28,6 @@ pydicom==2.4.4
 pydicom-seg==0.4.1
 pynetdicom==2.0.2
 pynrrd==1.0.0
-opencv-python-headless==4.9.0.80
```
Comment (Nic-Ma) [inline requirements.txt:31]: Hi @tangy5 , @SachidanandAlle , do we still need to update some tutorials or docs to let users install the OpenCV package as it's optional package now.

Thanks.

**Comment by SachidanandAlle** [comment]
Yes.. I agree.. we. Need to mention it.. or add optional dependency logic for installation 

pip install monailabel [opencv]

@tangy5

**Pull request merged**